server/app/views/admin/programs/ProgramPredicateConfigureView.java

```diff
@@ -524,8 +525,23 @@ private DivTag createScalarDropdown(
       }
     }
 
+    if (questionDefinition.isName()) {
```

Comment:
**suggestion (blocking):** It seems like this if-condition would fit better on line 519.  So, could you do this?  
```
if (questionDefinition.isAddress()) {
      scalars = ImmutableSet.of(Scalar.SERVICE_AREA);
      maybeSelectedScalar = Optional.of(Scalar.SERVICE_AREA);
    } else {
      try {
        scalars = Scalar.getScalars(questionDefinition.getQuestionType());
        if (questionDefinition.isName()) {
            // Filter out NAME_SUFFIX
            scalars =
                scalars.stream()
                    .filter(scalar -> !scalar.equals(Scalar.NAME_SUFFIX))
                    .collect(toImmutableSet());
        }
        maybeSelectedScalar =
            assertLeafOperationNode(maybeLeafNode).map(LeafOperationExpressionNode::scalar);
      } catch (InvalidQuestionTypeException | UnsupportedQuestionTypeException e) {
        // This should never happen since we filter out Enumerator questions before this point.
        return div()
            .withText("Sorry, you cannot create a show/hide predicate with this question type.");
      }
    }